packages/grafana-ui/src/components/Layout/Box/Box.tsx

```diff
@@ -0,0 +1,214 @@
+export type BorderColor = keyof GrafanaTheme2['colors']['border'] | 'error' | 'success' | 'warning' | 'info';
+export type BorderRadius = keyof ThemeShape['radius'];
+export type BoxShadow = keyof ThemeShadows;
+
+interface BoxProps {
+  // Margin props
+  /** Sets the property `margin` */
+  margin?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `margin-top` and `margin-bottom`. Higher priority than margin. */
+  marginX?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `margin-left` and `margin-right`. Higher priority than margin. */
+  marginY?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-top`. Higher priority than margin and marginY. */
+  marginTop?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-bottom`. Higher priority than margin and marginXY */
+  marginBottom?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-left`. Higher priority than margin and marginX. */
+  marginLeft?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-right`. Higher priority than margin and marginX. */
+  marginRight?: ResponsiveProp<ThemeSpacingTokens>;
+
+  // Padding props
+  /** Sets the property `padding` */
+  padding?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `padding-top` and `padding-bottom`. Higher priority than padding. */
+  paddingX?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `padding-left` and `padding-right`. Higher priority than padding. */
+  paddingY?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-top`. Higher priority than padding and paddingY. */
+  paddingTop?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-bottom`. Higher priority than padding and paddingY. */
+  paddingBottom?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-left`. Higher priority than padding and paddingX. */
+  paddingLeft?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-right`. Higher priority than padding and paddingX. */
+  paddingRight?: ResponsiveProp<ThemeSpacingTokens>;
+
+  // Border Props
+  borderStyle?: ResponsiveProp<BorderStyle>;
+  borderColor?: ResponsiveProp<BorderColor>;
+  borderRadius?: ResponsiveProp<BorderRadius>;
+
+  // Flex Props
+  /** Sets the property `flex` */
+  grow?: ResponsiveProp<number>;
+  /** Sets the property `flex-shrink` */
+  shrink?: ResponsiveProp<number>;
+  alignItems?: ResponsiveProp<AlignItems>;
+  justifyContent?: ResponsiveProp<JustifyContent>;
+
+  // Other props
+  backgroundColor?: ResponsiveProp<BackgroundColor>;
+  display?: ResponsiveProp<Display>;
+  boxShadow?: ResponsiveProp<BoxShadow>;
+  /** Sets the HTML element that will be rendered as a Box. Defaults to 'div' */
+  element?: ElementType;
+}
+
+export const Box = ({ children, ...props }: React.PropsWithChildren<BoxProps>) => {
+  const styles = useStyles2(useCallback((theme) => getStyles(theme, props), [props]));
```

Comment:
should move this to the new useStyles and ensure memoisation still occurs correctly.

Reply:
I thought this was the new `useStyles`? 🤔  What should I change it to?

Reply:
Oh sorry - you missed the big news from yesterday https://github.com/grafana/grafana/pull/75000

you should be able to rewrite this as `const styles = useStyles2(getStyles, props)`. I _think_ the props object should be shallow-ly compared for memoisation, which would be good to double check.

Reply:
Ooh nice!

Reply:
I did try having just `props` there but it was not being compared correctly so I extracted all the props